python/cugraph/cugraph/testing/resultset.py

```diff
@@ -108,7 +109,11 @@ def load_resultset(resultset_name, resultset_download_url):
         if not compressed_file_path.exists():
             urllib.request.urlretrieve(resultset_download_url, compressed_file_path)
         tar = tarfile.open(str(compressed_file_path), "r:gz")
-        tar.extractall(str(curr_resultset_download_dir))
+        # TODO: pass filter="fully_trusted" when minimum supported Python version >=3.12
+        #  ref: https://docs.python.org/3/library/tarfile.html#tarfile-extraction-filter
+        with warnings.catch_warnings():
```

Comment:
Starting with Python 3.12, using `tar.extractall()` without passing the `filter` argument raises this `DeprecationWarning`:

```text
DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.
```

That warning is treated as an error for CI here, so I added this warning filter to allow these tests to pass.

([build link](https://github.com/rapidsai/cugraph/actions/runs/10781109765/job/29911632066))